eden_server.py

```diff
@@ -74,43 +74,43 @@ def get_models(config):
 )
 def run(config):
 
-    config['weight_decay'] = 0.05 + 0.1 * random.random()
-    config['learning_rate'] = 0.05 + 0.05 * random.random()
-    config['lr_decay_rate'] = 0.96 + 0.039 * random.random()
-    config['cutn'] = [48, 36, 30]
-    config['batch_size'] = 1
-    config['text_inputs'] = get_permuted_prompts(config['text_input'], 2)
+    config.data['weight_decay'] = 0.05 + 0.1 * random.random()
+    config.data['learning_rate'] = 0.05 + 0.05 * random.random()
+    config.data['lr_decay_rate'] = 0.96 + 0.039 * random.random()
+    config.data['cutn'] = [48, 36, 30]
+    config.data['batch_size'] = 1
+    config.data['text_inputs'] = get_permuted_prompts(config.data['text_input'], 2)
     print("config: \n", config)
 
     model, perceptor, preprocess = get_models(config = config)
-    augmentations = get_augmentations(config, config['__gpu__'])
+    augmentations = get_augmentations(config, config.gpu)
 
     #model = taming_transformers.model
-    model.post_quant_conv = model.post_quant_conv.to(config['__gpu__'])
-    model.decoder = model.decoder.to(config['__gpu__'])
+    model.post_quant_conv = model.post_quant_conv.to(config.gpu)
+    model.decoder = model.decoder.to(config.gpu)
 
     try:
-        width, height = config['width'], config['height']
-        octave_scale, num_octaves = config['octave_scale'], config['num_octaves']
-        progress = config['__progress__']
+        width, height = config.data['width'], config.data['height']
+        octave_scale, num_octaves = config.data['octave_scale'], config.data['num_octaves']
+        progress = config.data['__progress__']
 
         '''
         This block uses a heuristic to estimate the progress_step_size at each octave
         '''
-        #total_progress = [n*(config['octave_scale']**(2*i)) for i, n in enumerate(config['num_iterations'])]
+        #total_progress = [n*(config.data['octave_scale']**(2*i)) for i, n in enumerate(config.data['num_iterations'])]
         iter_per_sec = [4.962, 3.826, 1.964]  # determined empirically for sizes 128, 256, 512
-        total_progress = [n/i for i, n in zip(iter_per_sec, config['num_iterations'])]
-        progress_step_sizes = [t/(n * sum(total_progress)) for n, t in zip(config['num_iterations'], total_progress)]
+        total_progress = [n/i for i, n in zip(iter_per_sec, config.data['num_iterations'])]
```

Comment:
`config.data` is required only where you want to modify/write data into the config. When reading values, you can just use: `config['num_iterations']`

Reply:
`config.data['num_iterations']` also works in the same exact way, so for now this should not be an issue